src/backend/distributed/commands/text_search.c

```diff
@@ -0,0 +1,804 @@
+	if (!HeapTupleIsValid(tup)) /* should not happen */
+	{
+		elog(ERROR, "cache lookup failed for text search configuration %u",
+			 tsconfigOid);
+	}
+	Form_pg_ts_config config = (Form_pg_ts_config) GETSTRUCT(tup);
+
+	AlterOwnerStmt *stmt = makeNode(AlterOwnerStmt);
+	stmt->objectType = OBJECT_TSCONFIGURATION;
+	stmt->object = (Node *) get_ts_config_namelist(tsconfigOid);
+	stmt->newowner = GetRoleSpecObjectForUser(config->cfgowner);
+
+	ReleaseSysCache(tup);
+	return list_make1(stmt);
+}
+
+
+List *
+get_ts_config_namelist(Oid tsconfigOid)
+{
+	HeapTuple tup = SearchSysCache1(TSCONFIGOID, ObjectIdGetDatum(tsconfigOid));
+	if (!HeapTupleIsValid(tup)) /* should not happen */
+	{
+		elog(ERROR, "cache lookup failed for text search configuration %u",
+			 tsconfigOid);
+	}
+	Form_pg_ts_config config = (Form_pg_ts_config) GETSTRUCT(tup);
+
+	char *schema = get_namespace_name(config->cfgnamespace);
+	char *configName = pstrdup(NameStr(config->cfgname));
+	List *names = list_make2(makeString(schema), makeString(configName));
+
+	ReleaseSysCache(tup);
+	return names;
+}
+
+
+static List *
+get_ts_dict_namelist(Oid tsdictOid)
+{
+	HeapTuple tup = SearchSysCache1(TSDICTOID, ObjectIdGetDatum(tsdictOid));
+	if (!HeapTupleIsValid(tup)) /* should not happen */
+	{
+		elog(ERROR, "cache lookup failed for text search dictionary %u", tsdictOid);
+	}
+	Form_pg_ts_dict dict = (Form_pg_ts_dict) GETSTRUCT(tup);
+
+	char *schema = get_namespace_name(dict->dictnamespace);
+	char *dictName = pstrdup(NameStr(dict->dictname));
+	List *names = list_make2(makeString(schema), makeString(dictName));
+
+	ReleaseSysCache(tup);
+	return names;
+}
+
+
+static Oid
+get_ts_config_parser(Oid tsconfigOid)
+{
+	HeapTuple tup = SearchSysCache1(TSCONFIGOID, ObjectIdGetDatum(tsconfigOid));
+	if (!HeapTupleIsValid(tup)) /* should not happen */
+	{
+		elog(ERROR, "cache lookup failed for text search configuration %u", tsconfigOid);
+	}
+	Form_pg_ts_config config = (Form_pg_ts_config) GETSTRUCT(tup);
+	Oid parserOid = config->cfgparser;
+
+	ReleaseSysCache(tup);
+	return parserOid;
+}
+
+
+static char *
+get_ts_parser_tokentype_name(Oid parserOid, int32 tokentype)
+{
+	TSParserCacheEntry *parserCache = lookup_ts_parser_cache(parserOid);
+	if (!OidIsValid(parserCache->lextypeOid))
+	{
+		elog(ERROR, "method lextype isn't defined for text search parser %u", parserOid);
+	}
+
+	/* take lextypes from parser */
+	LexDescr *tokenlist = (LexDescr *) DatumGetPointer(
+		OidFunctionCall1(parserCache->lextypeOid, Int32GetDatum(0)));
+
+	/* and find the one with lexid = tokentype */
+	int j = 0;
+	while (tokenlist && tokenlist[j].lexid)
+	{
+		if (tokenlist[j].lexid == tokentype)
+		{
+			return pstrdup(tokenlist[j].alias);
+		}
+		j++;
+	}
+
+	/* we haven't found the token */
+	ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					errmsg("token type \"%d\" does not exist in parser", tokentype)));
+}
+
+
+static List *
+GetTSParserNameList(Oid tsparserOid)
```

Comment:
revisit name of function
```suggestion
get_ts_parser_namelist(Oid tsparserOid)
```